packages/pluggableWidgets/gallery-native/src/components/Gallery.tsx

```diff
@@ -0,0 +1,107 @@
+import { createElement, ReactElement, ReactNode } from "react";
+import { Text, FlatList, Pressable, View } from "react-native";
+import { ObjectItem, DynamicValue } from "mendix";
+import DeviceInfo from "react-native-device-info";
+import { GalleryStyle } from "../ui/Styles";
+import { PaginationEnum, ScrollDirectionEnum } from "../../typings/GalleryProps";
+import { isAvailable } from "@mendix/piw-utils-internal";
+
+export interface GalleryProps<T extends ObjectItem> {
+    emptyPlaceholder?: ReactNode;
+    hasMoreItems: boolean;
+    itemRenderer: (
+        renderWrapper: (children: ReactNode, className?: string, onClick?: () => void) => ReactElement,
+        item: T
+    ) => ReactElement;
+    items: T[] | undefined;
+    loadMoreItems: () => void;
+    filters?: ReactNode;
+    name: string;
+    pagination: PaginationEnum;
+    loadMoreButtonCaption?: DynamicValue<string>;
+    phoneColumns: number;
+    pullDown?: () => void;
+    pullDownIsExecuting?: boolean;
+    scrollDirection: ScrollDirectionEnum;
+    style: GalleryStyle;
+    tabletColumns: number;
+}
+
+export const Gallery = <T extends ObjectItem>(props: GalleryProps<T>): ReactElement => {
+    const isScrollDirectionVertical = props.scrollDirection === "vertical";
+    const numOfColumns = DeviceInfo.isTablet() ? props.tabletColumns : props.phoneColumns;
+    const firstItemId = props.items?.[0]?.id;
+    const lastItemId = props.items?.[props.items.length - 1]?.id;
+
+    const onEndReached = (): void => {
+        if (props.pagination === "virtualScrolling" && props.hasMoreItems) {
+            props.loadMoreItems();
+        }
+    };
+
+    const renderItem = (item: { item: T }): ReactElement =>
+        props.itemRenderer(
+            (children, className, onClick) => (
+                <Pressable
+                    style={isScrollDirectionVertical && { width: `${100 / numOfColumns}%` }}
+                    testID={`${props.name}-list-item-${item.item.id}`}
+                    onPress={onClick}
```

Comment:
What happens when the child doesnt have an action? Does it feel pressable in Android with Ripple effect?